app/services/settings-v2/video.ts

```diff
@@ -3,80 +3,201 @@ import { Inject } from 'services/core/injector';
+      fpsType: context.video.fpsType,
+      fpsCom: `${context.video.fpsNum}-${context.video.fpsDen}`,
+      fpsNum: context.video.fpsNum,
+      fpsDen: context.video.fpsDen,
+      fpsInt: context.video.fpsNum,
+    };
+  }
+
+  get baseResolution() {
+    const context = this.contexts.horizontal;
     return {
-      baseRes: `${context.baseWidth}x${context.baseHeight}`,
-      outputRes: `${context.outputWidth}x${context.outputHeight}`,
-      scaleType: context.scaleType,
-      fpsType: context.fpsType,
-      fpsCom: `${context.fpsNum}-${context.fpsDen}`,
-      fpsNum: context.fpsNum,
-      fpsDen: context.fpsDen,
-      fpsInt: context.fpsNum,
+      width: context.video.baseWidth,
+      height: context.video.baseHeight,
     };
   }
 
+  get isGreen(): boolean {
+    return false;
+  }
+
   get videoSettings() {
-    return this.settingsManagerService.videoSettings;
+    return this.greenService.views.videoSettings;
   }
 
-  migrateSettings() {
-    Object.keys(this.videoSettings).forEach((key: keyof obs.IVideo) => {
-      this.SET_VIDEO_SETTING(key, this.videoSettings[key]);
-    });
+  migrateSettings(display?: TDisplayType) {
+    if (display === 'default' || display === 'horizontal') {
+      this.state.horizontal = this.contexts.horizontal.video;
+      Object.keys(this.contexts.horizontal.legacySettings).forEach(
+        (key: keyof obs.IAdvancedStreaming | keyof obs.ISimpleStreaming) => {
+          this.SET_VIDEO_SETTING(key, this.contexts.horizontal.legacySettings[key]);
+        },
+      );
+      Object.keys(this.contexts.horizontal.video).forEach((key: keyof obs.IVideo) => {
+        this.SET_VIDEO_SETTING(key, this.contexts.horizontal.video[key]);
+        if (
+          invalidFps(this.contexts.horizontal.video.fpsNum, this.contexts.horizontal.video.fpsDen)
+        ) {
+          this.createDefaultFps();
+        }
+      });
+    } else {
+      const data = this.videoSettings.green;
 
-    if (invalidFps(this.state.videoContext.fpsNum, this.state.videoContext.fpsDen)) {
-      this.setVideoSetting('fpsNum', 30);
-      this.setVideoSetting('fpsDen', 1);
+      this.state[display] = this.contexts[display].video;
+      Object.keys(data).forEach(
+        (key: keyof obs.IAdvancedStreaming | keyof obs.ISimpleStreaming) => {
+          this.SET_VIDEO_SETTING(key, data[key], display);
+          if (
+            invalidFps(this.contexts[display].video.fpsNum, this.contexts[display].video.fpsDen)
+          ) {
+            this.createDefaultFps(display);
+          }
+        },
+      );
     }
   }
 
-  establishVideoContext() {
-    if (this.state.videoContext) return;
+  establishVideoContext(display: TDisplayType = 'horizontal') {
+    if (this.contexts[display]) return;
+
+    this.contexts[display] = obs.VideoFactory.create();
+    this.state[display] = {} as obs.IVideoInfo;
+    this.migrateSettings(display);
+    this.contexts[display].video = this.state[display];
```

Comment:
i don't think this is necessary, since curing the migration we already set these values to mirror eachother

Reply:
Agreed. Removed